bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,881 @@
+            Thread.currentThread().interrupt();
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/comm-error.error-in-sysap-com");
+
+            throw new FreeAtHomeHttpCommunicationException(0,
+                    "Http communication interrupted [ " + e.getMessage() + " ]");
+        } catch (ExecutionException | TimeoutException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/comm-error.error-in-sysap-com");
+
+            throw new FreeAtHomeHttpCommunicationException(0,
+                    "Http communication interrupted [ " + e.getMessage() + " ]");
+        }
+
+        return true;
+    }
+
+    /**
+     * Method to process socket events
+     */
+    public void setDatapointOnWebsocketFeedback(String receivedText) {
+        JsonReader reader = new JsonReader(new StringReader(receivedText));
+        reader.setLenient(true);
+        JsonElement jsonTree = JsonParser.parseReader(reader);
+
+        // check the output
+        if (jsonTree.isJsonObject()) {
+            JsonObject jsonObject = jsonTree.getAsJsonObject();
+
+            jsonObject = jsonObject.getAsJsonObject(sysApUID);
+            jsonObject = jsonObject.getAsJsonObject("datapoints");
+
+            Set<String> keys = jsonObject.keySet();
+
+            Iterator<String> iter = keys.iterator();
+
+            while (iter.hasNext()) {
+                String eventDatapointID = iter.next();
+
+                JsonElement element = jsonObject.get(eventDatapointID);
+                String value = element.getAsString();
+
+                // channelUpdateHandler.updateChannelByDatapointEvent(eventDatapointID, value);
+
+                String[] parts = eventDatapointID.split("/");
+
+                FreeAtHomeDeviceHandler deviceHandler = mapEventListeners.get(parts[0]);
+
+                if (deviceHandler != null) {
+                    deviceHandler.onDeviceStateChanged(eventDatapointID, value);
+                }
+
+                logger.debug("Socket event processed: event-datapoint-ID {} value {}", eventDatapointID, value);
+            }
+        }
+    }
+
+    public void registerDeviceStateListener(String deviceID, FreeAtHomeDeviceHandler deviceHandler) {
+        mapEventListeners.put(deviceID, deviceHandler);
+    }
+
+    public void unRegisterDeviceStateListener(String deviceID) {
+        mapEventListeners.remove(deviceID);
+    }
+
+    /**
+     * Method to open Http connection
+     */
+    public boolean openHttpConnection() {
+        boolean ret = false;
+
+        try {
+            // Add authentication credentials.
+            AuthenticationStore auth = httpClient.getAuthenticationStore();
+
+            URI uri1 = new URI(baseUrl);
+            auth.addAuthenticationResult(new BasicAuthentication.BasicResult(uri1, username, password));
+
+            String url = baseUrl + "/rest/devicelist";
```

Comment:
You have the string "/rest/devicelist" in line 141 as well, perhaps it would be better to keep this as a class constant to avoid this duplication?

Reply:
**Fixed**